internal/provider/kubernetes/kubernetes.go

```diff
@@ -73,6 +79,12 @@ func New(cfg *rest.Config, svr *config.Server, resources *message.ProviderResour
 		return nil, fmt.Errorf("unable to set up ready check: %w", err)
 	}
 
+	// Emit elected & continue with deployment of infra resources
```

Comment:
are we blocking here ?
imo we should continue to process watching resources, translating them, and populating the xds cache
we should only block on updating status

Reply:
This won't be blocking as the manager is configured with:
Controller:              config.Controller{NeedLeaderElection: ptr.To(false)}
So as expected, we process watching resources, translating them, and populating the xds cache
and only block on updating status and the k8s envoy proxy provider (that can actually deploy envoys) if not elected.

Reply:
cool can we update the comment here - "Wait until election is complete, and then start reconciling resources"

Reply:
so then the question I have is, I dont see any code that is disabling status writes, how is this magically working :) ?

Reply:
almost, actually the status controller is already good with:
https://github.com/envoyproxy/gateway/blob/a667f23602794a38195cf0318a17e735bfcd33a1/internal/status/status.go#L106

Reply:
ah so the updates via the status updater client is auto disabled  ?

Reply:
yes its a runnable, the manager won't call its start unless its a leader.
the client update won't be triggered.

type Runnable interface {
	// Start starts running the component.  The component will stop running
	// when the context is closed. Start blocks until the context is closed or
	// an error occurs.
	Start(context.Context) error
}